content/blog/eng/careful-with-that-constant-developer.md

```diff
@@ -0,0 +1,462 @@
+{
+    public const SIDE_OF_THE_FORCE = 'dark';
+}
+
+$obiwan = new Master();
+echo $obiwan->useTheForce(); 
+// light
+$anakin = new Padawan();
+echo $anakin->useTheForce(); 
+// light
+```
+
+It appears that the super class (in which the method *useTheForce()* belongs) is able to keep unchanged its constant even when the derived class uses it.  
+But what happens if you make a small change to the previous example?   
+Try to change the access to the constant by replacing `self` with `static` this way
+
+```php
+<?php
+declare(strict_types=1);
+
+interface JediInterface 
+{
+    public const SIDE_OF_THE_FORCE = 'light';
+    
+    public function useTheForce(): string;
+}
+
+class Master implements JediInterface 
+{
+    public function useTheForce(): string 
+    {
+        return static::SIDE_OF_THE_FORCE;
+    } 
+}
+
+class Padawan extends Master 
+{
+    public const SIDE_OF_THE_FORCE = 'dark';
+}
+
+$obiwan = new Master();
+echo $obiwan->useTheForce(); 
+// light
+$anakin = new Padawan();
+echo $anakin->useTheForce(); 
+// dark
+```
+
+That's how late static bindings feature works.  
+`self`, being a static reference to the current class, is resolved using the class in which the method belongs.  
+On the other hand, late static bindings feature with the keyword `static` goes beyond that limitation, by referencing the class that was initally called at runtime.  
+Is it safe to use late stating bindings with constants?  
+Again, it's probably a matter of approach. Constants shouldn't be allowed to change, but in case, be sure that the things you do, can't reveal unpleasant surprises. If you expect to get the light side of the force and you get the dark side, you could be disappointed.
+
+# Namespace constants
+
+Outside of a class hierarchy, PHP allows you to define constants in a [namespace context](https://www.php.net/manual/en/language.namespaces.basics.php) too.  
+Indeed you can do something like this  
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace Jedi {
+    const SIDE_OF_THE_FORCE = 'light';
+}
+
+namespace Sith {
+    const SIDE_OF_THE_FORCE = 'dark';
+}
+
+namespace {
+    echo Jedi\SIDE_OF_THE_FORCE;
+    // light
+    echo Sith\SIDE_OF_THE_FORCE;
+    // dark
+}
+```
+
+Notice the unusual namespace bracketed syntax, it's the only way for combining into a single file global non-namespaced code with namespaced code. Anyway, this example, it's just for demonstration purposes, because it's strongly discouraged as a coding practice to combine multiple namespaces into the same file.  
+Notice also that since you are out of any class hierarchy, you are not allowed to use access modifiers for `const` that is therefore set to a public default visibility. 
+
+# Sensing the future
+
+All the scenarios we've briefly seen until now could be addressed differently, thinking of one of the new PHP features.  
+Although other languages featured Enumerations for a long time, they are going to be available in PHP only since version 8.1.  
```

Comment:
Although other languages have supported Enumerations for a long time, they will only be available in PHP from version 8.1.